modules/ROOT/pages/design-common-errors-08.adoc

```diff
@@ -0,0 +1,360 @@
+      "colors" : null ,
+      "label" : "Community" ,
+      "url" : "/home/home.jsp?tsid=02uxx00000056Sw"
+    } , {
+      "type" : "Tabset" ,
+      "content" : null ,
+      "icons" : null ,
+      "colors" : null ,
+      "label" : "App Launcher" ,
+      "url" : "/app/mgmt/applauncher/appLauncher.apexp?tsid=02uxx00000056Sx",
+    }
+  ]
+}
+----
+
+== Common Error 3
+// APIMF-852 and 853
+
+
+*_Using a non-integer data type for the example of a property that is an integer_*
+
+The data type for the example of a property that is an integer must be an integer. As an example of the error, this snippet shows that the data type of the example does not match the data type defined for `unit_price`
+
+
+----
+#%RAML 0.8
+title: ExampleRAML
+
+/products:
+  /product/{product_id}:
+    description: Get product instance details
+    get:
+      responses:
+        200:
+          body:
+            application/json:
+              schema: |
+                {
+                  "$schema": "http://json-schema.org/draft-03/schema",
+                  "id": "http://jsonschema.net",
+                  "properties": {
+                      "unit_price": {
+                          "type": "integer"
+                      }
+                  }
+                }
+              example: |
+                {
+                  "unit_price": 19.00
+                }
+----
+
+As another example of the error, the data type of the example does not match the data type defined for `uriParameters`.
+
+
+----
+#%RAML 0.8
+title: Order Entry API
+/orders:
+  /{id}:
+    uriParameters:
+      id:
+        type: integer
+        example: "451"
+----
+
+== Common Error 4
+// APIMF-896
+
+*_Not using a required property in the example of a schema that defines that property_*
+
+An example of a schema that defines a required property must use that property. In this snippet, the property that is defined in the second `id` attribute is misspelled: `sfOrderId` should be `sOrderId`. The second snippet shows the example JSON response in which `sOrderId` is defined. The example, therefore, is not in fact using the required property because the property was not defined correctly.
+
+
+----
+#%RAML 0.8
+title: ExampleRAML
+/order:
+  /{id}:
+    get:
+      responses:
+        200:
+          body:
+            application/json:
+              schema: |
+                {
+                  "type":"object",
+                  "$schema": "http://json-schema.org/draft-03/schema",
+                  "id": "http://com.mulesoft.demo.orders.get.json.order",
+                  "properties":{
+                      "property": {
+                        "type":"string",
+                        "id": "http://com.mulesoft.demo.orders.create.json.get.sfOrderId",
+                        "required":true
+                      }
+                  }
+                }
+              example: |
+                {
+                  "anotherProp": 14523
+                }
+----
+
```

Comment:
The error is going to be that the mandatory property called "property" is missing.

Reply:
I must have misunderstood this comment in the ticket:
_The backward compatibility issue here is that in 0.8, java parser is not validating the missing property that is marked as required. In the original api that is attached, you can see in get_order_response.json that there is a typo in "sOrderId" (should be "sfOrderId"), so the requirement of that property is not being fullfiled._
Then, the error is that the example should use the key `property`, not the key `anotherProp`?

Reply:
I think the anotherProp is ok, because the example is trying to show that there is a missing required property at the example.

Reply:
I've rewritten the description of this error. Please have a look.

Reply:
I've corrected the write-up of this error.